presto-main/src/main/java/com/facebook/presto/operator/scalar/MapTransformValueFunction.java

```diff
@@ -237,7 +238,16 @@ private static MethodHandle generateTransform(Type keyType, Type valueType, Type
                 .body(new BytecodeBlock()
                         .append(loadKeyElement)
                         .append(loadValueElement)
-                        .append(transformedValueElement.set(function.invoke("apply", Object.class, keyElement.cast(Object.class), valueElement.cast(Object.class)).cast(transformedValueJavaType)))
+                        .append(
+                                new TryCatch(
+                                        "Close builder before throwing to avoid subsequent calls finding it in an inconsistent state if we are in in a TRY() call.",
+                                        transformedValueElement.set(function.invoke("apply", Object.class, keyElement.cast(Object.class), valueElement.cast(Object.class))
+                                                .cast(transformedValueJavaType)),
+                                        new BytecodeBlock()
+                                                .append(mapBlockBuilder.invoke("closeEntry", BlockBuilder.class).pop())
+                                                .append(pageBuilder.invoke("declarePosition", void.class))
+                                                .throwObject(),
```

Comment:
nit: To have the same behavior as in `map_zip_with`: https://github.com/prestodb/presto/blob/9266c01533fc0562b99cd47c0d2815e34958a399/presto-main/src/main/java/com/facebook/presto/operator/scalar/MapZipWithFunction.java#L174-L181, you can do the following:
```java
                                        new BytecodeBlock()
                                                .append(mapBlockBuilder.invoke("closeEntry", BlockBuilder.class).pop())
                                                .append(pageBuilder.invoke("declarePosition", void.class))
                                                .putVariable(transformationException)
                                                .append(invokeStatic(Throwables.class, "throwIfUnchecked", void.class, transformationException))
                                                .append(newInstance(RuntimeException.class, transformationException))
                                                .throwObject(),
```

So it also generates bytecode to call `throwIfUnchecked` and throw the exception wrapped in `RuntimeException`. 

Here is the whole code block:

```java
        Variable transformationException = scope.declareVariable(Throwable.class, "transformationException");
        body.append(new ForLoop()
                .initialize(position.set(constantInt(0)))
                .condition(lessThan(position, positionCount))
                .update(incrementVariable(position, (byte) 2))
                .body(new BytecodeBlock()
                        .append(loadKeyElement)
                        .append(loadValueElement)
                        .append(
                                new TryCatch(
                                        "Close builder before throwing to avoid subsequent calls finding it in an inconsistent state if we are in in a TRY() call.",
                                        transformedValueElement.set(function.invoke("apply", Object.class, keyElement.cast(Object.class), valueElement.cast(Object.class))
                                                .cast(transformedValueJavaType)),
                                        new BytecodeBlock()
                                                .append(mapBlockBuilder.invoke("closeEntry", BlockBuilder.class).pop())
                                                .append(pageBuilder.invoke("declarePosition", void.class))
                                                .putVariable(transformationException)
                                                .append(invokeStatic(Throwables.class, "throwIfUnchecked", void.class, transformationException))
                                                .append(newInstance(RuntimeException.class, transformationException))
                                                .throwObject(),
                                        type(Throwable.class)))
                        .append(keySqlType.invoke("appendTo", void.class, block, position, blockBuilder))
                        .append(writeTransformedValueElement)));
```